src/yandex_cloud_ml_sdk/_datasets/dataset.py

```diff
@@ -136,6 +142,68 @@ async def _delete(
 
         logger.info("Dataset %s successfully deleted", self.id)
 
+    @safe_on_delete
+    async def _download(
+        self,
+        *,
+        download_path: PathLike,
+        timeout: float = 60,
+    ) -> tuple[Path, ...]:
+        logger.debug("Downloading dataset %s", self.id)
+
+        base_path = coerce_path(download_path)
+        if not base_path.exists():
+            raise ValueError(f"{base_path} does not exist")
+
+        if not base_path.is_dir():
+            raise ValueError(f"{base_path} is not a directory")
+
+        if os.listdir(base_path):
```

Comment:
[Path.iterdir()](https://docs.python.org/3/library/pathlib.html#pathlib.Path.iterdir)

I mean, you should choose pathlib or no and stick with it